kod/object/active/flag.kod

```diff
@@ -907,6 +911,23 @@ messages:
       return;
    }
 
+   HalfStaff(duration=86400000)
+   "for mourning purposes, in game or not default duratio is 1 day 86400000"
+   "Max duration is Kod Max int 134217728(A little more than 37 hours)"
+   {
+      piFlagHotSpot = 4;
+      ptHalfStaffOff = CreateTimer(self,@HalfStaffTimer,duration);
+
+      return;
+   }
+
+   HalfStaffTimer()
+   {
+      piFlagHotSpot = 1;
```

Comment:
HalfStaffTimer should take a parameter named timer (default $). The timer parameter is sent by blakserv when a timer fires, so if this parameter is $ it means the message was called manually and it should delete ptHalfStaffOff if it isn't $.
Also set ptHalfStaffOff to $ here (otherwise saving the game will give a timer reference error).